src/container/disk/hash/disk_extendible_hash_table.cpp

```diff
@@ -4,131 +4,156 @@
 //
 // disk_extendible_hash_table.cpp
 //
-// Identification: src/container/hash/extendible_hash_table.cpp
+// Identification: src/container/disk/hash/disk_extendible_hash_table.cpp
 //
-// Copyright (c) 2015-2021, Carnegie Mellon University Database Group
+// Copyright (c) 2015-2023, Carnegie Mellon University Database Group
 //
 //===----------------------------------------------------------------------===//
 
+#include "container/disk/hash/disk_extendible_hash_table.h"
+
 #include <iostream>
 #include <string>
 #include <utility>
 #include <vector>
 
+#include "common/config.h"
 #include "common/exception.h"
 #include "common/logger.h"
+#include "common/macros.h"
 #include "common/rid.h"
-#include "container/disk/hash/disk_extendible_hash_table.h"
+#include "common/util/hash_util.h"
+#include "storage/index/hash_comparator.h"
+#include "storage/page/extendible_htable_bucket_page.h"
+#include "storage/page/extendible_htable_directory_page.h"
+#include "storage/page/extendible_htable_header_page.h"
+#include "storage/page/page_guard.h"
 
 namespace bustub {
 
-template <typename KeyType, typename ValueType, typename KeyComparator>
-HASH_TABLE_TYPE::DiskExtendibleHashTable(const std::string &name, BufferPoolManager *buffer_pool_manager,
-                                         const KeyComparator &comparator, HashFunction<KeyType> hash_fn)
-    : buffer_pool_manager_(buffer_pool_manager), comparator_(comparator), hash_fn_(std::move(hash_fn)) {
-  //  implement me!
+template <typename K, typename V, typename KC>
+DiskExtendibleHashTable<K, V, KC>::DiskExtendibleHashTable(const std::string &name, BufferPoolManager *bpm,
+                                                           const KC &cmp, HashFunction<K> hash_fn,
```

Comment:
it's a little bit weird to have const & for `cmp` but `HashFunction` without reference, but it looks fine overall